rs/ic_os/config/src/setupos/mod.rs

```diff
@@ -1,2 +1,194 @@
 pub mod config_ini;
 pub mod deployment_json;
+
+use config_types::*;
+use deployment_json::VmResources;
+use macaddr::MacAddr6;
+use url::Url;
+
+pub fn create_setupos_config(
+    node_reward_type: Option<String>,
+    mgmt_mac: MacAddr6,
+    deployment_environment: DeploymentEnvironment,
+    nns_urls: &[Url],
+    vm_resources: VmResources,
+    enable_trusted_execution_environment: bool,
+    use_nns_public_key: bool,
+    use_node_operator_private_key: bool,
+    use_ssh_authorized_keys: bool,
+    verbose: bool,
+    network_settings: NetworkSettings,
+) -> SetupOSConfig {
+    let icos_settings = ICOSSettings {
+        node_reward_type,
+        mgmt_mac,
+        deployment_environment,
+        logging: Logging {},
+        use_nns_public_key,
+        nns_urls: nns_urls.to_vec(),
+        use_node_operator_private_key,
+        enable_trusted_execution_environment,
+        use_ssh_authorized_keys,
+        icos_dev_settings: ICOSDevSettings::default(),
+    };
+
+    let setupos_settings = SetupOSSettings;
+
+    // Only allow choosing VM memory for dev.
+    #[cfg(feature = "dev")]
+    let memory = vm_resources.memory.unwrap_or(PROD_GUEST_VM_MEMORY);
+
+    #[cfg(not(feature = "dev"))]
+    let memory = PROD_GUEST_VM_MEMORY;
```

Comment:
`memory` from the deployment JSON is now optional, and `vm_memory` in `HostOSSettings` is still always set. If `memory` is specified on `dev`, that value is used, else `PROD_GUEST_VM_MEMORY`.

I'd like to take it further and always have `HostOSSettings` read back `PROD_GUEST_VM_MEMORY` on `prod`, but that's difficult while we still access struct fields directly.